### PR TITLE
6438 API debtor business search. Also some work on building detail re…

### PR DESCRIPTION
**Pull request by doug-lovett**
*Issue #:* /bcgov/entity#6438

*Description of changes:*
API debtor business search. Also some work on building detail results JSON.

Signed-off-by: Doug Lovett <doug@diamante.ca>

### CODE REVIEW

File: ppr-api/README.md [hunk 1]
```diff
@@ -23,13 +23,43 @@ GitHub Pages (https://guides.github.com/features/pages/) are a neat way to docum
...
 
+### Mock pay-api environment variable.
+To use the mock pay-api service for local testing, set the .env variable:
+   PAYMENT_SVC_URL="https://bcregistry-bcregistry-mock.apigee.net/pay/api/v1"
+
 ### Development Setup
-1. Open the ppr-api directory in VS Code to treat it as a project (or WSL projec). To prevent version clashes, set up avirtual environment to install the Python packages used by this project.
+1. Open the ppr-api directory in VS Code to treat it as a project (or WSL projec). To prevent version clashes, set up a virtual environment to install the Python packages used by this project.
 1. Run `make setup`
 1. Run `pip install .`
```
Comment (thorwolpert) [inline ppr-api/README.md:64]: this should be included in the Makefile, if not, let me know and I'll update it.

File: ppr-api/src/ppr_api/models/client_party.py [hunk 2]
```diff
@@ -12,32 +12,25 @@
...
 from __future__ import annotations
 
-from enum import Enum
-
-#from sqlalchemy import event
-
 from .db import db
-
-from .address import Address  # noqa: F401 pylint: disable=unused-import; needed by the SQLAlchemy relationship
+from .address import Address  # noqa: F401 pylint: disable=unused-import
```
Comment (thorwolpert) [inline ppr-api/src/ppr_api/models/client_party.py:22]: we try and put in the reason for the override after a **;**

File: ppr-api/src/ppr_api/models/db.py [hunk 3]
```diff
@@ -16,12 +16,13 @@
 These will get initialized by the application using the models
 """
 from flask_sqlalchemy import SQLAlchemy
-#from sqlalchemy_continuum import make_versioned
```
Comment (thorwolpert) [inline ppr-api/src/ppr_api/models/db.py:19]: you can remove all the continuum bits from the models, as PPR wont use it for versioning

File: ppr-api/src/ppr_api/models/financing_statement.py [hunk 4]
```diff
@@ -12,25 +12,24 @@
...
 # pylint: disable=too-many-statements, too-many-branches
 
 from __future__ import annotations
 
 from enum import Enum
 from http import HTTPStatus
 
-#from sqlalchemy import event
-
+# from sqlalchemy import event
```
Comment (thorwolpert) [inline ppr-api/src/ppr_api/models/financing_statement.py]: you can just remove these.
when Sonarcloud is hooked up to this, it'll complain about commented out code.

File: ppr-api/src/ppr_api/models/financing_statement.py [hunk 5]
```diff
@@ -12,25 +12,24 @@
...
 from ppr_api.exceptions import BusinessException
 from ppr_api.models import utils as model_utils
 
 from .db import db
-from .registration import Registration  # noqa: F401 pylint: disable=unused-import; needed by the SQLAlchemy relationship
-from .trust_indenture import TrustIndenture  # noqa: F401 pylint: disable=unused-import; needed by the SQLAlchemy relationship
-from .party import Party  # noqa: F401 pylint: disable=unused-import; needed by the SQLAlchemy relationship
-from .general_collateral import GeneralCollateral  # noqa: F401 pylint: disable=unused-import; needed by the SQLAlchemy relationship
-from .vehicle_collateral import VehicleCollateral  # noqa: F401 pylint: disable=unused-import; needed by the SQLAlchemy relationship
+from .registration import Registration  # noqa: F401 pylint: disable=unused-import
```
Comment (thorwolpert) [inline ppr-api/src/ppr_api/models/financing_statement.py]: we keep the comments as to why a lint was disabled

File: ppr-api/src/ppr_api/models/search_client.py [hunk 6]
```diff
@@ -12,19 +12,19 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """This module holds model data and database operations for search queries."""
+# flake8: noqa Q000,E122,E131
```
Comment (thorwolpert) [inline ppr-api/src/ppr_api/models/search_client.py:15]: please add comments as to why the lint exclusions?

File: ppr-api/src/ppr_api/models/search_client.py [hunk 7]
```diff
@@ -233,23 +269,57 @@ def search_by_serial_type(self):
...
+                self.search_response = json.dumps(results_json)
+        else:
+            self.returned_results_size = 0
+            self.total_results_size = 0
 
     def search(self):
-        """Execute a search by the previously set search type."""
-
-        if self.search_type_cd == 'RG':
+        """Execute a search by the previously set search type and criteria."""
```
Comment (thorwolpert) [inline ppr-api/src/ppr_api/models/search_client.py:310]: the exception from save just keeps floating up?

File: ppr-api/src/ppr_api/models/search_result.py [hunk 8]
```diff
@@ -11,22 +11,22 @@
...
+"""This module holds model data and database operations for search results detail requests."""
 
 from __future__ import annotations
 
 from http import HTTPStatus
 import json
-import copy
 
-#from sqlalchemy import event
+# from sqlalchemy import event
```
Comment (thorwolpert) [inline ppr-api/src/ppr_api/models/search_result.py]: you can remove these

File: ppr-api/src/ppr_api/models/trust_indenture.py [hunk 9]
```diff
@@ -14,12 +14,8 @@
 """This module holds data for financing statement trust indenture information."""
 from __future__ import annotations
 
-#from sqlalchemy import event
-
-#from ppr_api.exceptions import BusinessException
-
 from .db import db
-from .registration import Registration  # noqa: F401 pylint: disable=unused-import; needed by the SQLAlchemy relationship
+from .registration import Registration  # noqa: F401 pylint: disable=unused-import
```
Comment (thorwolpert) [inline ppr-api/src/ppr_api/models/trust_indenture.py:19]: we like to keep the comments as to why the linter is suppressed

File: ppr-api/src/ppr_api/services/payment/__init__.py [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module wraps the calls to external payment service used by the API."""
+import os
+
+
+# Service endpoint including /api/v1 with or without trailing backslash.
+PAYMENT_SVC_URL = os.getenv('PAYMENT_SVC_URL')
```
Comment (thorwolpert) [inline ppr-api/src/ppr_api/services/payment/__init__.py]: this is a bad pattern for flask that crept into the other services, so don't follow it.
should come from , current_app.config.get('PAYMENT_SVC_URL')

File: ppr-api/src/ppr_api/services/payment/client/__init__.py [hunk 11]
```diff
@@ -0,0 +1,241 @@
...
+import os
+
+from flask import current_app
+
+import requests
+
+# Follow env variable names from other API's
+# Service endpoint with or without trailing backslash.
+PAYMENT_SVC_URL = os.getenv('PAYMENT_SVC_URL')
+PAYMENT_SVC_PREFIX = os.getenv('PAYMENT_SVC_PREFIX', 'api/v1/')
```
Comment (thorwolpert) [inline ppr-api/src/ppr_api/services/payment/client/__init__.py]: this is a bad pattern for flask that crept into the other services, so don't follow it.
should come from , current_app.config.get('PAYMENT_SVC_URL')

File: ppr-api/tests/unit/api/test_amendment.py [hunk 12]
```diff
@@ -103,7 +105,7 @@ def test_amendment_valid_CO_200(session, client, jwt):
     assert rv.status_code == HTTPStatus.OK
 
 
-def test_amendment_valid_AM_200(session, client, jwt):
+def test_amendment_valid_am_200(session, client, jwt):
```
Comment (thorwolpert) [inline ppr-api/tests/unit/api/test_amendment.py:108]: Could these be table driven tests?

**Review by thorwolpert — CHANGES_REQUESTED** [review]
Some changes to look at.
for tests, maybe look at leveraging: pytest.mark.parametrize

File: ppr-api/README.md (hunk 1 above)
Comment (doug-lovett) [reply ppr-api/README.md:64]: The makefile also has setup-venv and install-dev.

File: ppr-api/src/ppr_api/models/client_party.py (hunk 2 above)
Comment (doug-lovett) [reply ppr-api/src/ppr_api/models/client_party.py:22]: With the comment flake8 generates:
src/ppr_api/models/client_party.py:21:1: I003 isort expected 1 blank line in imports, found 0
src/ppr_api/models/client_party.py:21:1: I003 isort expected 1 blank line in imports, found 0
I'll put the comment in a separate line.

File: ppr-api/src/ppr_api/models/db.py (hunk 3 above)
Comment (doug-lovett) [reply ppr-api/src/ppr_api/models/db.py:19]: Continnuum references and versioning removed.

File: ppr-api/src/ppr_api/models/financing_statement.py (hunk 4 above)
Comment (doug-lovett) [reply ppr-api/src/ppr_api/models/financing_statement.py]: Removed.

File: ppr-api/src/ppr_api/models/financing_statement.py (hunk 5 above)
Comment (doug-lovett) [reply ppr-api/src/ppr_api/models/financing_statement.py]: Comments restored. Initially they were generating flake8 messages.

File: ppr-api/src/ppr_api/models/search_result.py (hunk 8 above)
Comment (doug-lovett) [reply ppr-api/src/ppr_api/models/search_result.py]: Removed.

File: ppr-api/src/ppr_api/models/trust_indenture.py (hunk 9 above)
Comment (doug-lovett) [reply ppr-api/src/ppr_api/models/trust_indenture.py:19]: When I restore the comment I see these flake8 messages:
src/ppr_api/models/trust_indenture.py:18:1: I003 isort expected 1 blank line in imports, found 0
src/ppr_api/models/trust_indenture.py:18:1: I003 isort expected 1 blank line in imports, found 0
src/ppr_api/models/trust_indenture.py:18:121: E501 line too long (121 > 120 characters)
Added comment as a separate line.

File: ppr-api/tests/unit/api/test_amendment.py (hunk 12 above)
Comment (doug-lovett) [reply ppr-api/tests/unit/api/test_amendment.py:108]: Yes, I'll keep that in mind as they are maintained. Maybe with https://docs.pytest.org/en/latest/parametrize.html

Comment (thorwolpert) [reply ppr-api/tests/unit/api/test_amendment.py:108]: yep, lots of examples about eg. https://github.com/bcgov/lear/blob/master/queue_services/entity-filer/tests/unit/test_configuration.py

File: ppr-api/src/ppr_api/models/search_client.py (hunk 7 above)
Comment (doug-lovett) [reply ppr-api/src/ppr_api/models/search_client.py:310]: Yes, it's just logging the occurrence and creating a specific message. The resource formats BusinessException responses.

File: ppr-api/src/ppr_api/models/search_client.py (hunk 6 above)
Comment (doug-lovett) [reply ppr-api/src/ppr_api/models/search_client.py:15]: Adding comments:
# Disable Q000: Allow query strings to be in double quotation marks that contain single quotation marks.
# Disable E122: allow query strings to be more human readable.
# Disable E131: allow query strings to be more human readable.

File: ppr-api/src/ppr_api/services/payment/__init__.py (hunk 10 above)
Comment (doug-lovett) [reply ppr-api/src/ppr_api/services/payment/__init__.py]: Remove: only required in the client.

File: ppr-api/src/ppr_api/services/payment/client/__init__.py (hunk 11 above)
Comment (doug-lovett) [reply ppr-api/src/ppr_api/services/payment/client/__init__.py]: Changed os.getenv to current_app.config.get.

**Review by thorwolpert — APPROVED** [review]

**Pull request merged**